packages/components/src/DataTable/Item/DataTableRow.tsx

```diff
@@ -119,6 +121,18 @@ const DataTableRowLayout = forwardRef(
 
 DataTableRowLayout.displayName = 'DataTableRowLayout'
 
+const getRowHoverColor = (
+  theme: DefaultTheme,
+  hasOnClick: boolean,
+  isHeader: boolean | undefined,
+  isSelected: boolean | 'mixed' | undefined
+) => {
+  if (!isHeader || !hasOnClick) return undefined
```

Comment:
Logic for `isHeader` here is backwards :) 


```suggestion
  if (isHeader || !hasOnClick) return undefined
```